### PR TITLE
Adding deleteMessage, deleteThread, archiveThread, and unarchiveThread endpoints.

### PR DESCRIPTION
**Pull request by MarkehMe**
This also adds message_id (mid) to message objects.

In my current project I required being able to run these methods (particularly deleteMessage and archiveThread).

Usage is fairly simple:

**deleteMessage**

    api.deleteMessage(message.message_id, function(err) {
      if(err) { 
        console.log('Error while deleting message..');
      } 
    });

**deleteThread**

    api.deleteThread(message.thread_id, function(err) {
      if(err) { 
        console.log('Error while deleting a thread..');
      } 
    });

**archiveThread**

    api.archiveThread(message.thread_id, function(err) {
      if(err) { 
        console.log('Error while archiving a thread..');
      } 
    });

**unarchiveThread**

    api.deleteMessage(thread_to_unarchive, function(err) {
      if(err) { 
        console.log('Error while unarchiving a thread..');
      } 
    });

### CODE REVIEW

File: src/archiveThread.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+"use strict";
+
+var utils = require("../utils");
+var log = require("npmlog");
+
+module.exports = function(mergeWithDefaults, api, ctx) {
+  return function archiveThread(thread_id, callback) {
+
+    var form = mergeWithDefaults();
+    form['ids['+thread_id+']'] = true;
```
Comment (bsansouci) [inline src/archiveThread.js]: Would this work with more than one thread? If so we might as well support an array of `thread_id`s too.

File: src/deleteMessage.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+var utils = require("../utils");
+var log = require("npmlog");
+
+module.exports = function(mergeWithDefaults, api, ctx) {
+  return function deleteMessage(message, callback) {
+    var message_id = message.message_id || message;
+    
+    var form = mergeWithDefaults({
+      'client' : 'mercury',
+      'message_ids[0]' : message_id
```
Comment (bsansouci) [inline src/deleteMessage.js]: Same as above ^

File: src/deleteThread.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+var utils = require("../utils");
+var log = require("npmlog");
+
+module.exports = function(mergeWithDefaults, api, ctx) {
+  return function deleteThread(thread_id, callback) {
+
+    var form = mergeWithDefaults({
+      'client' : 'mercury',
+      'ids[0]' : thread_id
```
Comment (bsansouci) [inline src/deleteThread.js]: Same :p

File: src/unarchiveThread.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+"use strict";
+
+var utils = require("../utils");
+var log = require("npmlog");
+
+module.exports = function(mergeWithDefaults, api, ctx) {
+  return function unarchiveThread(thread_id, callback) {
+
+    var form = mergeWithDefaults();
+    form['ids['+thread_id+']'] = false;
```
Comment (bsansouci) [inline src/unarchiveThread.js]: Ok last one ;)

**Comment by bsansouci** [comment]
Wow huge PR! Thank you so much for this.

**Comment by MarkehMe** [comment]
Supporting an array would be an awesome idea, don't know why I didn't think to do it - I'll add it now!

**Comment by bsansouci** [comment]
This looks good, is there anything you want to add @MarkehMe ?

**Comment by MarkehMe** [comment]
This is all for now, I can't think of anything else in regards to deleting messages at the moment. 

Maybe I could extend the message object here: https://github.com/Schmavery/facebook-chat-api/blob/master/utils.js#L113-L122 and add .delete() and .archive() functions onto it?

**Comment by MarkehMe** [comment]
I'll attempt to resolve conflicts for you now

**Comment by MarkehMe** [comment]
I created a conflict, if you want I can create a new pull request and just copy my changes over.

**Comment by bsansouci** [comment]
Mmmh I see no benefice to attaching functions to this kind of data, the simple and pure functions are good. We lose simple serializability with this too. 
Can you rebase and fix? What's the conflict?

**Comment by MarkehMe** [comment]
There you go. This removes the .delete and .archiveThread functions on the message object!

**Comment by bsansouci** [comment]
Ok thanks for this!

**Pull request merged**